test/proto/perf_control.proto

```diff
@@ -0,0 +1,154 @@
+// Copyright 2015, Google Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// An integration test service that covers all the method signature permutations
+// of unary/streaming requests/responses.
+syntax = "proto3";
+
+import "test/proto/messages.proto";
+import "test/proto/perf_stats.proto";
+
+package grpc.testing;
+
+enum ClientType {
+  SYNCHRONOUS_CLIENT = 0;
+  ASYNC_CLIENT = 1;
+}
+
+enum ServerType {
+  SYNCHRONOUS_SERVER = 0;
+  ASYNC_SERVER = 1;
+}
+
+enum RpcType {
+  UNARY = 0;
+  STREAMING = 1;
+}
+
+enum LoadType {
+  CLOSED_LOOP = 0;
+  POISSON = 1;
+  UNIFORM = 2;
+  DETERMINISTIC = 3;
+  PARETO = 4;
+}
+
+message PoissonParams {
+  double offered_load = 1;
+}
+
+message UniformParams {
+  double interarrival_lo = 1;
+  double interarrival_hi = 2;
+}
+
+message DeterministicParams {
+  double offered_load = 1;
+}
+
+message ParetoParams {
+  double interarrival_base = 1;
+  double alpha = 2;
+}
+
+message LoadParams {
+  oneof load {
+    PoissonParams poisson = 1;
+    UniformParams uniform = 2;
+    DeterministicParams determ = 3;
+    ParetoParams pareto = 4;
+  };
+}
+
+message ClientConfig {
+  repeated string server_targets = 1;
+  ClientType client_type = 2;
+  bool enable_ssl = 3;
```

Comment:
I will add the field now. This is a different situation from compress where
a default value of false is ok if we decide to add it later.


On Tue, Nov 3, 2015 at 11:36 AM Jan Tattermusch <notifications@github.com>
wrote:

> In test/proto/perf_control.proto
> <https://github.com/grpc/grpc/pull/3905#discussion_r43795865>:
>
> > +  double alpha = 2;
> > +}
> > +
> > +message LoadParams {
> > +  oneof load {
> > +    PoissonParams poisson = 1;
> > +    UniformParams uniform = 2;
> > +    DeterministicParams determ = 3;
> > +    ParetoParams pareto = 4;
> > +  };
> > +}
> > +
> > +message ClientConfig {
> > +  repeated string server_targets = 1;
> > +  ClientType client_type = 2;
> > +  bool enable_ssl = 3;
>
> The problem with not including use_test_ca field here is that if we later
> reintroduce it, it will have a default value of false (bcuz proto3), which
> is in conflict with our current test semantics (where we are testing with
> use_test_ca = true by default).
>
> Besides this point, I don't feel strongly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/3905/files#r43795865>.
>